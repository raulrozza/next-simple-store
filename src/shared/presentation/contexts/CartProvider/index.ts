export { default as useAddToCart } from './hooks/useAddToCart';
