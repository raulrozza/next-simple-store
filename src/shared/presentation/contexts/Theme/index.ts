export { default as useTheme } from './hooks/useTheme';
export { default as Theme } from './implementations/StyledComponentTheme';
