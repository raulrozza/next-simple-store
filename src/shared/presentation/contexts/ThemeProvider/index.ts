export { default as useTheme } from './hooks/useTheme';
export { default as ThemeProvider } from './implementations/StyledComponentTheme';
