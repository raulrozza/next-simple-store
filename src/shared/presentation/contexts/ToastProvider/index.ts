export { useToastProvider } from './context';
export { default as ToastProvider } from './implementations/ReactToastifyProvider';
