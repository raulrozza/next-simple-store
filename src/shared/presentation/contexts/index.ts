export * from './CartProvider';
export * from './ThemeProvider';
