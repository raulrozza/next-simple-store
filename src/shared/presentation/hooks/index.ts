import { trpc } from '@/shared/infra/http/trpc';

const { useQuery, useMutation } = trpc;

export { useQuery, useMutation };
