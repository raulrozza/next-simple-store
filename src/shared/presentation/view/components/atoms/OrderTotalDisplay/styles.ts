import styled, { css } from 'styled-components';

export const Container = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: space-around;
`;

interface TotalTextProps {
    isValidTotal: boolean;
}

export const TotalText = styled.p<TotalTextProps>`
    ${({ theme, isValidTotal }) => css`
        font-size: ${theme.typography.sizes.lg};

        ${!isValidTotal &&
        css`
            color: ${theme.palette.error['500']};
        `}
    `}
`;
