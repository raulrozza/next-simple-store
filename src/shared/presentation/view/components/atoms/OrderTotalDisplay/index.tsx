import { FC } from 'react';

import { Container, TotalText } from './styles';

interface OrderTotalDisplayProps {
    title: string;
    totalText: string;
    isTotalValid: boolean;
}

const OrderTotalDisplay: FC<OrderTotalDisplayProps> = ({
    title,
    totalText,
    isTotalValid,
}) => (
    <Container>
        <h3>{title}</h3>

        <TotalText
            isValidTotal={isTotalValid}
            data-testid="orderTotalDisplay-TotalText"
        >
            {totalText}
        </TotalText>
    </Container>
);

export default OrderTotalDisplay;
