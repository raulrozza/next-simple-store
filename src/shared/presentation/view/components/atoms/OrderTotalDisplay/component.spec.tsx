import { render, screen } from '@testing-library/react';

import theme from '@/config/theme';
import { ThemeProvider } from '@/shared/presentation/contexts';

import OrderTotalDisplay from './index';

describe('OrderTotalDisplay', () => {
    it('should render the component displaying the text correctly, without the color style on total', () => {
        const view = render(
            <OrderTotalDisplay
                title="Total display"
                totalText="$10.00"
                isTotalValid={true}
            />,
            {
                wrapper: ThemeProvider,
            },
        );

        expect(screen.getByText('Total display')).toBeInTheDocument();
        expect(screen.getByText('$10.00')).toBeInTheDocument();
        expect(screen.getByText('$10.00')).not.toHaveStyle({
            color: theme.palette.error['500'],
        });
        expect(view).toMatchSnapshot();
    });

    it('should render the total error colored if the total is invalid', () => {
        const view = render(
            <OrderTotalDisplay
                title="Total display"
                totalText="$10.00"
                isTotalValid={false}
            />,
            {
                wrapper: ThemeProvider,
            },
        );

        expect(screen.getByText('$10.00')).toHaveStyle({
            color: theme.palette.error['500'],
        });
        expect(view).toMatchSnapshot();
    });
});
