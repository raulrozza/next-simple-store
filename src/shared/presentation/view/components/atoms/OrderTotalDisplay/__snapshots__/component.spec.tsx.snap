// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderTotalDisplay should render the component displaying the text correctly, without the color style on total 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c1 {
  font-size: 1.5rem;
}

<body>
    <div>
      <div
        class="c0"
      >
        <h3>
          Total display
        </h3>
        <p
          class="c1"
          data-testid="orderTotalDisplay-TotalText"
        >
          $10.00
        </p>
      </div>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c1 {
  font-size: 1.5rem;
}

<div>
    <div
      class="c0"
    >
      <h3>
        Total display
      </h3>
      <p
        class="c1"
        data-testid="orderTotalDisplay-TotalText"
      >
        $10.00
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`OrderTotalDisplay should render the total error colored if the total is invalid 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c1 {
  font-size: 1.5rem;
  color: #CD6B55;
}

<body>
    <div>
      <div
        class="c0"
      >
        <h3>
          Total display
        </h3>
        <p
          class="c1"
          data-testid="orderTotalDisplay-TotalText"
        >
          $10.00
        </p>
      </div>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c1 {
  font-size: 1.5rem;
  color: #CD6B55;
}

<div>
    <div
      class="c0"
    >
      <h3>
        Total display
      </h3>
      <p
        class="c1"
        data-testid="orderTotalDisplay-TotalText"
      >
        $10.00
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
